### ISSUE TITLE
Issues following guide - can't start VM

### ISSUE THREAD

**Issue by thebozzcl** [Arc-Compute/libvf.io#12]
Hi, I'm testing this out (love the project!) and I'm having issues following the guide's steps after installation.

I can create my initial VM using `arcd create /home/bozz/libvfio/vm/nvidia-mdev.yaml /home/bozz/libvfio/iso/Win10_21H1_English_x64.iso 400`. After Windows is installed, I shut down the VM (I've tried both shutting it down directly from inside and via `arcd stop $UUID`). Then I run `arcd start /home/bozz/libvfio/vm/nvidia-mdev.yaml --preinstall --save`, but nothing happens.

```
bozz@Snek:~/libvfio$ arcd ps
+--------------------------------------+-------+-------------+--------+---------+--------------------------------+--------+-----------+
| UUID                                 | PID   | Kernel      | States | GPU/NET | Ports                          | S/C/T  | RAM Alloc |
+--------------------------------------+-------+-------------+--------+---------+--------------------------------+--------+-----------+
| 2b698a0b-e72a-46b0-be17-3c96ae01e6fa | 14160 | windows.arc |        | 0/0     | @[(guest: 22, host: 2222)](+0) | 1/12/1 | 49152 MiB |
| 598df73f-f97c-44ba-b198-df69efa25008 | 13380 |             |        | 0/0     | @[(guest: 22, host: 2222)](+0) | 1/12/1 | 49152 MiB |
---------------------------------------+-------+-------------+--------+---------+--------------------------------+--------+------------
```

I poked around `~/.local/libvf.io/` and noticed the following:
* The kernel file for `2b698a0b-e72a-46b0-be17-3c96ae01e6fa` is in `./live/`, though I'd assume it would be in `./kernel/` after stopping the VM.
* There's not much going on in the logs... the QEMU logs were empty and the ARCD logs didn't have useful information.
* The lock file for the apparently failed VMs are still there even though the VM didn't run.
* I checked on `mdevctl` and I noticed each new execution defined a new device instead of re-using the one from the base implementation. I assume that's not correct, since `mdevctl` seems to only be able to allocate one device.
* Running the QEMU command directly failed with a `no such file or directory` for the new VM's live kernel file.

Looks to me like copying the kernel files is not quite working, but I don't want to jump to conclusions since I'm not very familiar with the project.

I was able to get the `--preinstall --save` step to work by grabbing the QEMU command from the console and replacing the UUIDs with the original VM and the vGPU, but it doesn't seem right, since it seems the idea would be to copy the kernel (which seems it's not happening) and re-create a vGPU (which might also not be working, since the original one wasn't destroyed and `mdevctl` reports 0 available GPUs of any type).

I'm still wrapping my head around QEMU, MDEV and your code so I'm not sure what to look for here, so I'd really appreciate the help. I'm more than happy to provide you with more details and try stuff as well.

BTW, any way I can enable additional logging/debugging on `arcd`? I didn't see an option for it while looking around in the code.

**Comment by thebozzcl** [comment]
I just noticed there's new commits, I'm gonna try pulling those in and see if it helps.

**Comment by arthurrasmusson** [comment]
@thebozzcl There were unstable changes introduced on November 3rd which were causing problems for me on my Arch box and also for some other folks on Ubuntu 20.04. I rolled them back today. @mbuchel and I will take a more detailed look at your issue in the morning but I would start with just what you suggested, trying the latest commits and seeing if your issue goes away.

**Comment by arthurrasmusson** [comment]
Also if you'd like to start your VM with no file copy use the `--no-copy` flag. That will launch the VM in place rather than copying it to the `/live/ `directory before booting it.

**Comment by arthurrasmusson** [comment]
>The lock file for the apparently failed VMs are still there even though the VM didn't run.

Ya, that's a bug we do need to patch. We had a daemon process called arcd-checkup that handled that on our legacy implementation. LibVF.IO is a ground up reimplemtation of that and some things like arcd-checkup haven't been ported over yet.


>I checked on mdevctl and I noticed each new execution defined a new device instead of re-using the one from the base implementation. I assume that's not correct, since mdevctl seems to only be able to allocate one device.
That

That's being worked on in feature/sudo-daemon but it isn't stable yet so we're not quite ready to merge it into master just yet. It's on the top of our list for things to fix however. :)

**Comment by arthurrasmusson** [comment]
>BTW, any way I can enable additional logging/debugging on arcd? I didn't see an option for it while looking around in the code.
I think that's a good idea, I'll try to add some sort of verbose logging mode tomorrow if I can get to it.

**Comment by thebozzcl** [comment]
Resolved by updating to the latest version. Something interesting I ran into though... if I just pulled the latest commits and re-ran the installation, my virtual machines ran super slowly. I solved it by formatting and starting from scratch.

Now I'm having issues with Looking Glass and my GPU drivers, but I don't think they're related to libvf.io and I don't have more details yet... so let's just close this!
